framework/src/outputs/formatters/SONDefinitionFormatter.C

```diff
@@ -0,0 +1,347 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SONDefinitionFormatter.h"
+#include "MooseUtils.h"
+#include "pcrecpp.h"
+
+SONDefinitionFormatter::SONDefinitionFormatter() : _spaces(2), _level(0) {}
+
+// ******************************** toString ******************************** //
+// TRAVERSE JSON ASSOCIATED TYPE ARRAY OF CPP_TYPES AND LOOKUP ABSOLUTE PATHS //
+// AND TRANSFORM THE PATHS TO WORK WITH OUR INPUT DOM HIERARCHY AND STORE FOR //
+// THE PARAMETER EXISTIN RULES AND ALSO STORE THE GLOBAL PARAMETERS JSON THEN //
+// CONSTRUCT BY ADDING ALL ROOT BLOCKS AND RETURN CONSTRUCTED STREAM'S STRING //
+// ************************************************************************** //
+std::string
+SONDefinitionFormatter::toString(const JsonVal & root)
+{
+  for (auto && type : root["global"]["associated_types"].getMemberNames())
```

Comment:
I don't because we don't use it anywhere else in the framework. Someday we may move to universal reference but it's almost never necessary and we've been avoiding it since many of our users aren't familiar enough with C++ to understand this more complex language features. I'd prefer to see them all dropped for consistency.